spec/feature/registration_from_another_application_spec.rb

```diff
@@ -112,14 +112,66 @@
     end
   end
 
+  context "when the feature_flag_enforce_levels_of_authentication is 'enabled'" do
+    before do
+      allow(Rails.configuration).to receive(:feature_flag_enforce_levels_of_authentication).and_return(true)
+    end
+
+    context "arriving from a level0 application" do
+      let(:application_scopes) { %i[openid level0 test_scope_read test_scope_write] }
+
+      it "completes the signup journey without MFA and stores :level_of_authentication level0" do
+        i_click_from_application
+        register_without_mfa
+
+        expect(page).to have_text(I18n.t("confirmation_sent.heading"))
+        expect(return_to_app_link(page)).to include("scope=openid+level0")
+      end
+    end
+
+    context "arriving from a level1 application" do
+      let(:application_scopes) { %i[openid level1 test_scope_read test_scope_write] }
+
+      it "completes the signup journey with MFA and stores :level_of_authentication level1" do
+        i_click_from_application
+        register_with_mfa
+
+        expect(page).to have_text(I18n.t("confirmation_sent.heading"))
+        expect(return_to_app_link(page)).to include("scope=openid+level1")
+      end
+    end
+  end
+
+  def return_to_app_link(page)
+    return_link_css_selector = ".govuk-link[data-module='explicit-cross-domain-links']"
+    html = Nokogiri.parse(page.body)
+    html.css(return_link_css_selector).first.attributes["href"].value
```

Comment:
Is this needed?  Capybara allows clicking on links and buttons by CSS selector, can we use that to find the element and extract its href?

Reply:
I tried `click_on I18n.t("confirmation_sent.continue_to_service", service_name: application.name)`
which results in:
```ruby
page.current_url
=> "http://localhost/sign-in"
```

I thought in this case we could try to test not just where the user ends up, but some idea of what their level of auth is?

Reply:
> some idea of what their level of auth is?

That would be something like `EphemeralState.last.level_of_authentication`